### PR TITLE
Override jacoco coverage

### PR DESCRIPTION
**Pull request by Deep1998**
Configure jacoco for spanner-common to exclude autovalue and exception classes in test coverage.
Also add unit test for null dialect check in avroTransform.

### CODE REVIEW

File: v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertorTest.java [hunk 1]
```diff
@@ -290,4 +290,27 @@ public void transformChangeEventTest_incorrectSpannerType() throws IOException {
...
+
+    GenericRecord genericRecord =
+        new GenericData.Record(
+            SchemaUtils.parseAvroSchema(
+                Files.readString(Paths.get("src/test/resources/avro/all-spanner-types.avsc"))));
+    genericRecord.put("bool_col", true);
+    GenericRecordTypeConvertor genericRecordTypeConvertor =
+        new GenericRecordTypeConvertor(mockSchemaMapper, "");
+
+    genericRecordTypeConvertor.transformChangeEvent(genericRecord, "all_types");
```
Comment (VardhanThigle) [inline v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertorTest.java:314]: Should we assert that mock was called at this point?

File: v2/spanner-common/pom.xml [hunk 2]
```diff
@@ -52,6 +52,73 @@
                         <phase>none</phase>
                     </execution>
                 </executions>
+            </plugin>
+		    <plugin>
```
Comment (darshan-sj) [inline v2/spanner-common/pom.xml:56]: Please correct indentation.

File: v2/spanner-common/pom.xml [hunk 3]
```diff
@@ -52,6 +52,73 @@
...
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>${jacoco.version}</version>
+                <configuration>
+                    <excludes>
+                        <!-- Excluding exception classes -->
+                        <exclude>**/*Exception.*</exclude>
+                        <!-- Excluding auto-generated classes. -->
+                        <exclude>**/*AutoValue_*</exclude>
+                    </excludes>
+                    <rules>
```
Comment (darshan-sj) [inline v2/spanner-common/pom.xml:67]: Line 67 to 81 is not needed. These inclusion and exclusions correspond to Spanner Import/Export templates.

File: v2/spanner-common/pom.xml [hunk 4]
```diff
@@ -52,6 +52,73 @@
...
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>default-report</id>
+                        <phase>prepare-package</phase>
+                        <goals>
+                            <goal>report</goal>
+                        </goals>
+                    </execution>
+                    <execution>
```
Comment (darshan-sj) [inline v2/spanner-common/pom.xml:97]: Also, Line 97 to 120 is not needed. This block is for coverage check and The coverage check is now handled through Codecov. Not needed here.

**Review by darshan-sj — APPROVED** [review]
Why is Unit tests failing for this PR? It had failed on the earlier commit as well.

**Review by VardhanThigle — APPROVED** [review]

**Comment by Deep1998** [comment]
> Why is Unit tests failing for this PR? It had failed on the earlier commit as well.

Earlier it failed becasue the method that it was testing was merged today morning only, rebased it today and it started working now (ignoring the transient failure).
